config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/PolicyEntityBuilder.java

```diff
@@ -469,6 +470,24 @@ private void prepareAssertion(Element policyElement, String assertionTag, Consum
         }
     }
 
+    @VisibleForTesting
+    void prepareMQRoutingAssertion(Bundle bundle, Element assertionElement) {
+        final Element activeConnectorNameElement = getSingleChildElement(assertionElement, ACTIVE_CONNECTOR_NAME, true);
+        if (activeConnectorNameElement != null) {
+            final String activeConnectorName = activeConnectorNameElement.getAttributes().getNamedItem(STRING_VALUE).getTextContent();
+            final SsgActiveConnector ssgActiveConnector = bundle.getSsgActiveConnectors().get(activeConnectorName);
+            final String id = ssgActiveConnector != null && ssgActiveConnector.getAnnotatedEntity() != null && ssgActiveConnector.getAnnotatedEntity().getId() != null ?
+                    ssgActiveConnector.getAnnotatedEntity().getId() : idGenerator.generate();
+            Element activeConnectorGoidElement = getSingleChildElement(assertionElement, ACTIVE_CONNECTOR_GOID, true);
+            Node activeConnectorGoidNode = activeConnectorGoidElement.getAttributes().getNamedItem(GOID_VALUE);
+            activeConnectorGoidNode.setTextContent(id);
+            Element activeConnectorIdElement = getSingleChildElement(assertionElement, ACTIVE_CONNECTOR_ID, true);
+            Node activeConnectorIdNode = activeConnectorIdElement.getAttributes().getNamedItem(GOID_VALUE);
```

Comment:
@uppoju , why are we not simplifying the policy i.e removing GOID from policy?

Reply:
i am just overwriting the value instead of removing and reading, not sure if it is OK

Reply:
we should remove GOID reference from policy during export and recreate while building the policy

Reply:
Done, to keep it in sync with other impl